conan/tools/cmake/cmakedeps.py

```diff
@@ -235,8 +235,9 @@ def join_paths_single_var(values):
         # Issue: #1251
         self.cxxflags_list = join_flags(";", cpp_info.cxxflags)
         self.cflags_list = join_flags(";", cpp_info.cflags)
-        self.sharedlinkflags_list = join_flags(";", cpp_info.sharedlinkflags)
-        self.exelinkflags_list = join_flags(";", cpp_info.exelinkflags)
+        self.sharedlinkflags_list = join_flags(";", ["-{}".format(f)
+                                                     for f in cpp_info.sharedlinkflags])
+        self.exelinkflags_list = join_flags(";", ["-{}".format(f) for f in cpp_info.exelinkflags])
```

Comment:
Could this break flags that already start with a `-` or `--`, for non-MSVC linkers?

Reply:
Yes, good catch, I will verify that too.

Reply:
Keeping flags that already start with "-" in https://github.com/conan-io/conan/pull/8740/commits/4ebe404bbd85c43c71a44626816dd419dc37c6b1